### ISSUE TITLE
[Bug Report] Weird Bug in Waterworld where agent cannot eat food

### ISSUE THREAD

**Issue by jjshoots** [Farama-Foundation/PettingZoo#923]
### Describe the bug

There's a weird bug in waterworld where sometimes an agent cannot eat food.

### Additional context

![render1_300_4afaae5e69758d1ed34c](https://user-images.githubusercontent.com/38184875/228069940-9af5e5fd-283a-4e31-9549-19a699570068.gif)

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/PettingZoo/issues) in the repo

**Comment by elliottower** [comment]
> I have a fix for this, but I noticed several other issues while looking into it. I'm putting together a PR

Thanks a lot for this, I looked into it myself at one point but it seemed pretty difficult to reproduce. Feel free to reach out via discord if you have any questions or anything, will take a look at the PR when you've created it

**Comment by elliottower** [comment]
Hi @dm-ackerman any chance you could submit the PR for this? Or if you are too busy could you describe what you did to fix this?

**Comment by dm-ackerman** [comment]
Yes, I can add that in a few days. It got delayed because of the warning items discussed on discord. I have those items too, but I have to wrap up another project today before I can post those.

**Comment by jkterry1** [comment]
Closed as waterworld is now removed
